TUM Campus App/LoginViewController.swift

```diff
@@ -13,29 +13,94 @@ class LoginViewController: UIViewController {
+            handleTextFieldInput(currentTextField: numbersTextField,
+                                 previousTextField: firstTextField,
+                                 nextTextField: secondTextField,
+                                 characterLimit: 2)
+        } else if sender === secondTextField {
+            handleTextFieldInput(currentTextField: secondTextField,
+                                 previousTextField: numbersTextField,
+                                 characterLimit: 3)
+        }
+
+        confirmButton.isEnabled = textFieldContentsAreValid()
+        confirmButton.alpha = textFieldContentsAreValid() ? 1 : 0.5
+    }
+
+    private func handleTextFieldInput(currentTextField: UITextField, previousTextField: UITextField? = nil, nextTextField: UITextField? = nil, characterLimit: Int) {
+        let text = currentTextField.text ?? ""
+
+        if text.count >= characterLimit {
+            currentTextField.text = String(text.prefix(characterLimit))
+
+            let substring = String(text.characters.dropFirst(characterLimit))
+            if let nextTextField = nextTextField {
+                nextTextField.becomeFirstResponder()
+                if !substring.isEmpty {
+                    nextTextField.text = substring
+                    textFieldEditingChanged(nextTextField)
+                }
+            } else {
+                currentTextField.resignFirstResponder()
+            }
+        } else if text.isEmpty {
+            guard let previousTextField = previousTextField else { return }
+
+            previousTextField.becomeFirstResponder()
+        }
+    }
+
+    func textFieldContentsAreValid() -> Bool {
+        guard let firstText = firstTextField.text, firstText.count == 2,
+            let numbersText = numbersTextField.text, numbersText.count == 2,
+            let secondText = secondTextField.text, secondText.count == 3 else { return false }
+
+        let isFirstTextValid = CharacterSet.letters.isSuperset(of: CharacterSet(charactersIn: firstText))
+        let isNumbersTextValid = Int(numbersText) != nil
+        let isSecondTextValid = CharacterSet.letters.isSuperset(of: CharacterSet(charactersIn: secondText))
+
+        highlightTextfield(textField: firstTextField, isHighlighted: !isFirstTextValid)
+        highlightTextfield(textField: numbersTextField, isHighlighted: !isNumbersTextValid)
+        highlightTextfield(textField: secondTextField, isHighlighted: !isSecondTextValid)
+
+        return isFirstTextValid && isNumbersTextValid && isSecondTextValid
     }
 
+    private func highlightTextfield(textField: UITextField, isHighlighted: Bool) {
+        textField.layer.borderWidth = isHighlighted ? 1 : 0
+        textField.layer.borderColor = isHighlighted ? UIColor.red.cgColor : UIColor(hexString: "0xC7C7CD").cgColor
+    }
 }
 
 extension LoginViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        firstTextField.delegate = self
```

Comment:
Did you set these through the Storyboard, or am I missing something?

Reply:
They are set through the storyboard. Thats why textFieldEditingChanged is an IBAction that receives textfield events ;)